src/lib/interfaces/drive_interface.cpp

```diff
@@ -93,8 +93,23 @@ static bool is_rotational_device(const std::string& device) {
     int is_rotational = 0;
     const auto maj_min{get_major_minor(device)};
     if (!maj_min.empty()) {
-        std::string sys_path = fmt::format("/sys/dev/block/{}/queue/rotational", maj_min);
-        if (auto rot_file = std::ifstream(sys_path); rot_file.is_open()) { rot_file >> is_rotational; }
+        char resolvedPath[PATH_MAX];
+        if (realpath(fmt::format("/sys/dev/block/{}", maj_min).c_str(), resolvedPath) == nullptr) {
```

Comment:
I don't want to force you to re-write this *again*, so I'll approve. But you may consider the more friendly: https://en.cppreference.com/w/cpp/filesystem/read_symlink
So you don't have to have statically sized arrays and such.

Reply:
no worry,  changing most to the filesystem and it seems cleaner, less fmt